### DISCUSSION TITLE
bidirectional cost function for semi-supervised training

### DISCUSSION THREAD

**Discussion by timroelofs123** [voxelmorph/voxelmorph#306] [Q&A] [answered]
Hello, I was wondering if it is possible to use a bidirectional cost function with the semi-supervised training with labels, i.e. through `VxmDenseSemiSupervisedSeg`? It does not appear to be supported as there is no `bidir` argument, though `VxmDenseSemiSupervisedPointCloud` does support it, so perhaps I am missing something.

**Comment by adalca** [comment, marked as answer]
@timroelofs123 thanks for your interest. You can pass `bidir` to `VxmDenseSemiSupervisedSeg` and it will be usedin the main (image-part) model, but you're right, we're not currently using it for the labels. This is probably a minor oversight when we re-launched vxm. I am tagging @ahoopes for this minor fix (or you @timroelofs123 can feel free to propose a pull request to the `dev` branch if you felt so inclinded :) ). 

Thanks!

**Reply by ahoopes** [reply]
Just added a `bidir_labels` flag (in 4f37e5f) that enables bidirectional image _and_ label warping.

**Reply by timroelofs123** [reply]
Awesome @ahoopes , thanks!
